sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -364,7 +369,132 @@ public String getDefaultLanguage() {
         return recognizeEntityAsyncClient.recognizeEntitiesBatch(documents, options);
     }
 
-    // Linked Entity
+    // PII Entity
+
+    /**
+     * Returns a list of Personally Identifiable Information(PII) entities in the provided document.
+     *
+     * For a list of supported entity types, check: <a href="https://aka.ms/tanerpii">this</a>.
+     * For a list of enabled languages, check: <a href="https://aka.ms/talangs">this</a>. This method will use the
+     * default language that sets up in {@link TextAnalyticsClientBuilder#defaultLanguage(String)}. If none is
+     * specified, service will use 'en' as the language.
+     *
+     * <p><strong>Code sample</strong></p>
+     * <p>Recognize the PII entities details in a document.
+     * Subscribes to the call asynchronously and prints out the recognized entity details when a response is
+     * received.</p>
+     *
+     * {@codesnippet com.azure.ai.textanalytics.TextAnalyticsAsyncClient.recognizePiiEntities#string}
+     *
+     * @param document The document to recognize PII entities details for.
+     * For text length limits, maximum batch size, and supported text encoding, see
+     * <a href="https://docs.microsoft.com/azure/cognitive-services/text-analytics/overview#data-limits">data limits</a>.
+     *
+     * @return A {@link Mono} contains a {@link PiiEntityCollection recognized PII entities collection}.
+     *
+     * @throws NullPointerException if {@code document} is null.
+     * @throws TextAnalyticsException if the response returned with an {@link TextAnalyticsError error}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<PiiEntityCollection> recognizePiiEntities(String document) {
```

Comment:
What happens here if the document is empty?

Reply:
api will make a TextDocumentInput with text = empty string, and call service. Service will return an batch with error result:
```
 if (entitiesResult.isError()) {
        throw logger.logExceptionAsError(toTextAnalyticsException(entitiesResult.getError()));
  }
```
It is the same behavior as other existing APIs

Reply:
hmm wondering if we know we already know its going to error, should we consider validating this at client side?
cc: @iscai-msft @maririos

Reply:
for single documents, in .NET we always validate that document is not null or empty. As an example: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs#L138

Not sure if Java is doing this for other endpoints, but worth checking and do the same we did in GA

Reply:
To be fair, for batch we check that documents is not null or empty, but the whole list not every element on it
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs#L248

Reply:
@maririos  I remember we don't check for empty document for user in single document but check for empty list for user.